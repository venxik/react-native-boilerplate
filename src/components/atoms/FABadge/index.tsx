import React, { memo } from 'react';

import { Badge } from 'native-base';

import type { IBadgeProps } from 'native-base';

export type IFABadgeProps = IBadgeProps;

export const FABadge = memo((props: IFABadgeProps) => {
  const { testID, ...baseProps } = props;

  return <Badge testID={testID} {...baseProps} />;
});
FABadge.displayName = 'FABadge';
