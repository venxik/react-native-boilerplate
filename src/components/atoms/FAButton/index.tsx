import React, { memo } from 'react';

import { Button } from 'native-base';

import type { IButtonProps } from 'native-base';

export type IFAButtonProps = IButtonProps;

export const FAButton = memo((props: IFAButtonProps) => {
  const { testID, ...baseProps } = props;

  return <Button testID={testID} {...baseProps} />;
});
FAButton.displayName = 'FAButton';
