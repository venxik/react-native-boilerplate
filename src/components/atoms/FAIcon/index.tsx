import React, { memo } from 'react';

import { Icon } from 'native-base';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Ionicons from 'react-native-vector-icons/Ionicons';

import type { IIconProps } from 'native-base';

export type IFAIconProps = {
  type?: 'font-awesome' | 'ionicons' | 'feather';
} & IIconProps;

export const FAIcon = memo((props: IFAIconProps) => {
  const { testID, type, ...baseProps } = props;

  let iconType: typeof FontAwesome;

  switch (type) {
    case 'ionicons':
      iconType = Ionicons;
      break;
    case 'feather':
      iconType = Feather;
      break;

    default:
      iconType = FontAwesome;
      break;
  }

  return <Icon testID={testID} as={iconType} {...baseProps} />;
});
FAIcon.displayName = 'FAIcon';
