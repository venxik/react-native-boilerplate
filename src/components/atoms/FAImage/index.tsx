import React, { memo } from 'react';

import { Image } from 'native-base';

import type { IImageProps } from 'native-base';

export type IFAImageProps = IImageProps;

export const FAImage = memo((props: IFAImageProps) => {
  const { testID, ...baseProps } = props;

  return <Image testID={testID} {...baseProps} />;
});
FAImage.displayName = 'FAImage';
