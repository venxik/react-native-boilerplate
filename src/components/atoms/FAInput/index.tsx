import React from 'react';
import { memo } from 'react';

import { Input, InputGroup, InputLeftAddon, InputRightAddon } from 'native-base';

import type { IInputProps } from 'native-base';

export type IFAInputProps = {
  leftAddon?: string | JSX.Element;
  rightAddon?: string | JSX.Element;
} & IInputProps;

export const FAInput = memo((props: IFAInputProps) => {
  const { testID, leftAddon, rightAddon, ...baseProps } = props;

  if (leftAddon || rightAddon) {
    return (
      <InputGroup>
        {leftAddon && <InputLeftAddon children={leftAddon} />}
        <Input testID={testID} {...baseProps} placeholder="nativebase" />
        {rightAddon && <InputRightAddon children={rightAddon} />}
      </InputGroup>
    );
  }

  return <Input testID={testID} {...baseProps} />;
});
FAInput.displayName = 'FAInput';
