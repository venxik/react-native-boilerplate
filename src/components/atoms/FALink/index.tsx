import React, { memo } from 'react';

import { Link } from 'native-base';

import type { ILinkProps } from 'native-base';

export type IFALinkProps = ILinkProps;

export const FALink = memo((props: IFALinkProps) => {
  const { testID, ...baseProps } = props;

  return <Link testID={testID} {...baseProps} />;
});
FALink.displayName = 'FALink';
