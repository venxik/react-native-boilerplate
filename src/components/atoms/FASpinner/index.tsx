import React, { memo } from 'react';

import { Heading, Spinner, Stack } from 'native-base';

import type { ISpinnerProps } from 'native-base';
import type { ColorType, ResponsiveValue } from 'native-base/lib/typescript/components/types';

export type IFASpinnerProps = {
  children?: string | number;
  color?: ColorType;
  direction?: ResponsiveValue<'row' | 'column' | 'column-reverse' | 'row-reverse'>;
} & ISpinnerProps;

export const FASpinner = memo((props: IFASpinnerProps) => {
  const { testID, children, color, size, direction, ...baseProps } = props;

  return (
    <Stack space={4} direction={direction ?? 'row'}>
      <Spinner testID={testID} {...baseProps} />
      {children && (
        <Heading color={color} fontSize={size}>
          {children}
        </Heading>
      )}
    </Stack>
  );
});
FASpinner.displayName = 'FASpinner';
