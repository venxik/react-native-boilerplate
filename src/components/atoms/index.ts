export * from './FABadge';
export * from './FAButton';
export * from './FAIcon';
export * from './FAImage';
export * from './FAInput';
export * from './FALink';
export * from './FARadio';
export * from './FASelect';
export * from './FASpinner';
export * from './FASwitch';
export * from './FATextArea';
