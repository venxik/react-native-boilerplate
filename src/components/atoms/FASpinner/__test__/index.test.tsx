import React from 'react';

import { FASpinner } from '@components/atoms';
import { render } from '@mocks';

describe('FASpinner', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<FASpinner>Test</FASpinner>);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with children prop', () => {
    const { getByTestId } = render(<FASpinner testID="spinner-example">Test</FASpinner>);

    expect(getByTestId('spinner-example')).toBeTruthy();
  });
});
