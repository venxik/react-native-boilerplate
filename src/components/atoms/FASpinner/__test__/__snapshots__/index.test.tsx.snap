// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FASpinner should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <ActivityIndicator
      accessibilityLabel="loading"
      accessible={true}
      color="#0891b2"
      size="small"
      style={
        Array [
          Array [
            Object {},
            Object {
              "dataSet": Object {},
            },
          ],
          undefined,
        ]
      }
    />
    <View
      dataSet={Object {}}
      style={
        Object {
          "width": 16,
        }
      }
    />
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": null,
          "fontStyle": "normal",
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 30,
          "textDecorationLine": undefined,
        }
      }
    >
      Test
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
